dotCMS/src/main/java/com/dotcms/util/ThreadContextUtil.java

```diff
@@ -0,0 +1,132 @@
+package com.dotcms.util;
+
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.UtilMethods;
+
+/**
+ * Thread Context Util provides methods to handle reindex stuff and other thread local context things
+ * @author jsancas
+ */
+public class ThreadContextUtil {
+
+    private static ThreadLocal<ThreadContext> contextLocal = new ThreadLocal<>();
+    private static final ThreadContextUtil     INSTANCE    = new ThreadContextUtil();
```

Comment:
ThreadContextUtil should be a static class - no reason to use a singleton pattern here.

Reply:
Done